indice_pollution/history/models/indice_uv.py

```diff
@@ -0,0 +1,130 @@
+                    bio.write(data)
+                filename = f'{today().strftime("%Y%m%d")}'
+                ftp.retrbinary(f'RETR {filename}.csv', callback=handle_binary)
+                b = bio.getvalue()
+                decoded_content = b.decode('UTF-8')
+            except Exception as e:
+                logging.error(e)
+                return None
+            return decoded_content
+
+        decoded_content = get_ftp_content()
+        if decoded_content is None:
+            return
+        reader = csv.DictReader(
+            decoded_content.splitlines(),
+            delimiter=';'
+        )
+        indices = []
+
+        def format_insee(raw_insee: str):
+            if len(raw_insee) == 5 and raw_insee.isdigit():
+                return raw_insee
+            else:
+                return None
+        def format_uv(raw_uv: str): # '-999' or '' means no value
+            if raw_uv.isdigit():
+                uv = int(raw_uv)
+                if uv < 0:
+                    uv = None
+            else:
+                uv = None
+            return uv
+
+        # file header: insee;commune;date;UV_J0;UV_J1;UV_J2;UV_J3
+        for row in reader:
+            insee = format_insee(row['insee'])
+            oudated_communes = []
+            if insee and insee not in oudated_communes:
+                departement_code = f'{insee:0>2}'
+                if departement_code == '20': # Corse: 20 in file, 2A or 2B in db
+                    insee_2a = insee.replace('20', '2A')
+                    commune = Commune.get(insee_2a)
+                    if commune is None:
+                        insee_2b = insee.replace('20', '2B')
+                        commune = Commune.get(insee_2b)
+                else:
+                    commune = Commune.get(insee)
+                date_format = '%Y-%m-%d'
+                try:
+                    date = datetime.strptime(row['date'], date_format).date()
+                except Exception as e:
+                    logging.error(e)
+                    continue
+                if commune and date:
+                    indices.append({
+                        **{
+                            "zone_id": commune.zone_id,
+                            "date": date,
+                            "uv_j0": format_uv(row['UV_J0']),
+                            "uv_j1": format_uv(row['UV_J1']),
+                            "uv_j2": format_uv(row['UV_J2']),
+                            "uv_j3": format_uv(row['UV_J3'])
+                        }
+                    })
+        ins = pg_insert(cls)\
+            .values(indices)\
+            .on_conflict_do_nothing()
+        db.session.execute(ins)
+        db.session.commit()
+
+    @classmethod
+    def get(cls, insee, date_=None):
+        date_ = date_ or today()
+        query = cls.query.join(Commune, Commune.zone_id == cls.zone_id).filter(Commune.insee == insee, IndiceUv.date==date_).order_by(IndiceUv.date.desc())
```

Comment:
À priori le `order_by` ne sert pas vu qu’on filtre sur la date
```suggestion
        query = cls.query.join(Commune, Commune.zone_id == cls.zone_id).filter(Commune.insee == insee, IndiceUv.date==date_)
```